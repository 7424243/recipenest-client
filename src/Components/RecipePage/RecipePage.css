h3 {
    padding-top: 20px;
    text-align: center;
    font-family: 'Baskervville', serif;
}

p {
    text-align: left;
    margin-left: 20%;
    margin-right: 20%;
}


/* Classes */

.buttons {
    margin-top: 50px;
    font-family: 'Baskervville', serif;
}

.recipe-page-container {
    text-align: center;
    padding-top: 5%;
    padding-bottom: 5%;
    font-family: 'Baskervville', serif;
    font-size: larger;
}

.recipe-page-img {
    margin-top: 20px;
    width: 30%;
    height: 30%;
    background-color: lightgray;
}