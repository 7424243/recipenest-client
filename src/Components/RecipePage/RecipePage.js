import React, {Component} from 'react'
import { Link } from 'react-router-dom'
import RecipenestContext from '../../RecipenestContext'
import './RecipePage.css'

class RecipePage extends Component {

    //allow access to context
    static contextType = RecipenestContext

    render() {
        
        const recipeId = parseInt(this.props.match.params.id)
        const {recipes} = this.context
        const getRecipe = (recipes, recipeId) =>
            recipes.find(recipe => recipe.id === recipeId)
        const recipeForPage = getRecipe(recipes, recipeId)

        return (
            <div className='recipe-page-container'>
                <div>
                    <img className='recipe-page-img'src={recipeForPage ? recipeForPage.img_url : null} alt='word cloud from recipe description'/>
                    <h3>{recipeForPage ? recipeForPage.recipe_name : null}</h3>
                    <a href={recipeForPage ? recipeForPage.url : null} target='_blank' rel='noreferrer'><h5>Clickable Recipe Link</h5></a>
                    <p>Description: {recipeForPage ? recipeForPage.description : null}</p>
                    <p>Additional Notes: {recipeForPage ? recipeForPage.notes : null}</p>
                </div>
                <div className='buttons'>
                    <Link to={`/edit/${this.props.match.params.id}`}><button>Edit</button></Link>
                    <button>Delete</button>
                    <Link to='/recipes'><button>Back</button></Link>
                </div>
            </div> 
        )
    }
}

export default RecipePage

