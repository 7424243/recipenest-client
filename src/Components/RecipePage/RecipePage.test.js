import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import RecipePage from './RecipePage'

describe('RecipePage Component', () => {

    //testing props
    const props = {
        match: {
            params: {
                id: 'Test_Recipe_Id'
            }
        }
    }

    //smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter><RecipePage {...props}/></BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    //snapshot test
    it('renders the UI as expected', () => {
        const tree = renderer 
          .create(<BrowserRouter><RecipePage {...props}/></BrowserRouter>)
          .toJSON()
        expect(tree).toMatchSnapshot()
      })
})
