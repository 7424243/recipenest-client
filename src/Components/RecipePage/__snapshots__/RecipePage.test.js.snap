// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipePage Component renders the UI as expected 1`] = `
<div
  className="recipe-page-container"
>
  <div>
    <img
      alt="word cloud from recipe description"
      className="recipe-page-img"
      src={null}
    />
    <h3 />
    <a
      href={null}
      rel="noreferrer"
      target="_blank"
    >
      <h5>
        Clickable Recipe Link
      </h5>
    </a>
    <p>
      Description: 
    </p>
    <p>
      Additional Notes: 
    </p>
  </div>
  <div
    className="buttons"
  >
    <a
      href="/edit/Test_Recipe_Id"
      onClick={[Function]}
    >
      <button>
        Edit
      </button>
    </a>
    <button>
      Delete
    </button>
    <a
      href="/recipes"
      onClick={[Function]}
    >
      <button>
        Back
      </button>
    </a>
  </div>
</div>
`;
